code/modules/mapping/mapping_helpers.dm

```diff
@@ -331,3 +331,73 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
 	. = ..()
 	var/area/A = get_area(get_turf(src))
 	A.color_correction = color_correction
+
+
+// This will put directional windows to adjucant turfs if airs will likely be vaccuumed.
+// Putting this on a space turf is recommended. If you put this on an open tile, it will place directional windows anyway.
+// If a turf is not valid to put a tile, it will automatically make a turf for failsafe.
+/obj/effect/mapping_helpers/space_window_placer
+	name = "Placer: Spaceproof directional windows"
+	icon_state = "directional_window_placer"
+	late = TRUE
+	var/window_type = /obj/structure/window/reinforced
+
+	var/static/list/unliable_atmos_blockers
+
+/obj/effect/mapping_helpers/space_window_placer/Initialize(mapload)
+	. = ..()
+	if(!unliable_atmos_blockers)
+		unliable_atmos_blockers = typecacheof(list(/obj/machinery/door))
+
+/obj/effect/mapping_helpers/space_window_placer/LateInitialize()
+	. = ..()
+	if(!z || !x || !y)
+		CRASH("It's not unable to place directional windows - xyz is null.")
+
+	var/turf/my_turf = get_turf(src)
+	var/list/nearby_turfs = list(
+		my_turf.get_nearby_turf_by_dir(SOUTH),
+		my_turf.get_nearby_turf_by_dir(NORTH),
+		my_turf.get_nearby_turf_by_dir(WEST),
+		my_turf.get_nearby_turf_by_dir(EAST))
```

Comment:
This is also not necessary, you can use get_adjacent_turfs or iterate over directions. There are already helpers to retrieve a list of turfs in each direction.

Reply:
the correct proc name was `get_adjacent_open_turfs` for that. now it makes sense, thanks.